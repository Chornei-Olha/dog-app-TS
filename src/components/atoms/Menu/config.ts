export const navMenuItems = [
  {
    text: 'Main',
    path: '/'
  },
  {
    text: 'Breeds',
    path: '/breeds'
  },
  {
    text: 'Vote',
    path: '/vote'
  },
  {
    text: 'History',
    path: '/history'
  },
  {
    text: 'Your uploads',
    path: '/your-uploads'
  }
];
