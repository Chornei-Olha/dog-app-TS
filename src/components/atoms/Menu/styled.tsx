import { styled } from '@mui/material/styles';
import { List, ListItem, ListItemProps, ListProps } from '@mui/material';

interface NavListProps extends ListProps {
  display?: string;
  bgcolor?: string;
}

export const NavList = styled(List)<NavListProps>(({ theme, display }) => ({
  display: display || 'flex',
  gap: '50px',
  backgroundColor: theme.palette.grey[600],
  padding: 0
}));

export const NavListItem = styled(ListItem)<ListItemProps>(() => ({
  padding: 0
}));
