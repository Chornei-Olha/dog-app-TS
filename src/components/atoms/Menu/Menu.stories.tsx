import type { Meta, StoryObj } from '@storybook/react';

import { Menu } from '.';
import { navMenuItems } from './config';

const meta = {
  title: 'Atom/Menu',
  component: Menu,
  argTypes: {},
  parameters: {
    layout: 'fullscreen'
  }
} satisfies Meta<typeof Menu>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Nav: Story = {
  args: {
    menuList: navMenuItems
  }
};
