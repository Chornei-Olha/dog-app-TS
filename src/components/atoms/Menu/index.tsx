import { Link } from '@mui/material';

import { NavList, NavListItem } from './styled';

type MenuItemProps = {
  text: string;
  path: string;
};

export type MenuProps = {
  menuList: MenuItemProps[];
};

export const Menu = ({ menuList }: MenuProps) => (
  <NavList component="nav">
    {menuList.map(item => (
      // react-router <Link/> add
      <NavListItem key={item.text}>
        <Link href={item.path} variant="h2" underline="none">
          {item.text}
        </Link>
      </NavListItem>
    ))}
  </NavList>
);
