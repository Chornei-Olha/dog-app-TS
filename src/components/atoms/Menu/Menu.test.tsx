import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';

import * as stories from './Menu.stories';

describe('Menu', () => {
  const { Nav } = composeStories(stories);
  it('should match snapshot Menu', () => {
    const { container } = render(<Nav />);
    expect(container).toMatchSnapshot();
  });
});
