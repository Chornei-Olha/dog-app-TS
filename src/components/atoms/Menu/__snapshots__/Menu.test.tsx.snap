// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot Menu 1`] = `
.emotion-0 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 50px;
  background-color: #757575;
  padding: 0;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  padding: 0;
}

.emotion-1.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-1.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-1.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-1.Mui-disabled {
  opacity: 0.38;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  font-size: 3.75rem;
  line-height: 1.2;
  letter-spacing: -0.00833em;
  color: #1976d2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <nav
    class="MuiList-root MuiList-padding emotion-0"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-h2 MuiLink-root MuiLink-underlineNone emotion-2"
        href="/"
      >
        Main
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-h2 MuiLink-root MuiLink-underlineNone emotion-2"
        href="/breeds"
      >
        Breeds
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-h2 MuiLink-root MuiLink-underlineNone emotion-2"
        href="/vote"
      >
        Vote
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-h2 MuiLink-root MuiLink-underlineNone emotion-2"
        href="/history"
      >
        History
      </a>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-1"
    >
      <a
        class="MuiTypography-root MuiTypography-h2 MuiLink-root MuiLink-underlineNone emotion-2"
        href="/your-uploads"
      >
        Your uploads
      </a>
    </li>
  </nav>
</div>
`;
